### DISCUSSION TITLE
FileUpload causing Array to string conversion error on create

### DISCUSSION THREAD

**Discussion by LemonPie22** [filamentphp/filament#7191] [Help] [answered]
In my `parts` resource I have the following FileUpload component:
```
FileUpload::make('documents')
    ->preserveFilenames()
    ->storeFileNamesIn('attachment_file_names')
    ->columnSpan('full')
    ->maxSize(1024)
    ->multiple()
    ->maxFiles(5)
    ->enableReordering()
    ->enableOpen(),
```
It works fine when I update records, including uploading files, but when I try to create a new part record (without any uploads) I'm getting an `Array to string conversion` error.

My `Part` model contains the following casts:
```
protected $casts = [
    'custom_fields' => 'array',
    'images' => 'array',
    'documents' => 'array',
];
```
Any hints please?

**Comment by danharrin** [comment, marked as answer]
`attachment_file_names` needs an array cast too

**Reply by LemonPie22** [reply]
Thank you @danharrin. That fixed it. I appreciate you taking the time during such a busy period.
